pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1776,15 +1776,20 @@ public void refreshSegment(String offlineTableName, SegmentMetadata segmentMetad
     sendSegmentRefreshMessage(offlineTableName, segmentName, true, true);
   }
 
-  public int reloadAllSegments(String tableNameWithType) {
-    LOGGER.info("Sending reload message for table: {}", tableNameWithType);
+  public int reloadAllSegments(String tableNameWithType, boolean forceDownload) {
+    LOGGER.info("Sending reload message for table: {} with forceDownload: {}", tableNameWithType, forceDownload);
+
+    if (forceDownload) {
+      TableType tt = TableNameBuilder.getTableTypeFromTableName(tableNameWithType);
+      Preconditions.checkArgument(tt == TableType.OFFLINE, "OFFLINE table is required to force segment download");
```

Comment:
Add the table name into the error message. Also add a TODO to support the realtime tables